app/assets/javascripts/discourse/app/pre-initializers/deprecate-jquery-integration.js

```diff
@@ -0,0 +1,46 @@
+import { assert, deprecate } from "@ember/debug";
+import EmberObject from "@ember/object";
+import Component from "@ember/component";
+import jQuery from "jquery";
+
+let done = false;
+
+// Adapted from https://github.com/emberjs/ember-jquery/blob/master/vendor/jquery/component.dollar.js
+// but implemented in a module to avoid transpiled version triggering the Ember Global deprecation.
+// To be dropped when we remove the jquery integration as part of the 4.x update.
+export default {
+  name: "deprecate-jquery-integration",
+
+  initialize() {
+    if (done) {
+      return;
+    }
+
+    EmberObject.reopen.call(Component, {
+      $(sel) {
+        assert(
+          "You cannot access this.$() on a component with `tagName: ''` specified.",
+          this.tagName !== ""
+        );
+
+        deprecate(
+          "Using this.$() in a component has been deprecated, consider using this.element",
+          false,
+          {
+            id: "ember-views.curly-components.jquery-element",
+            since: "3.4.0",
+            until: "4.0.0",
+            url: "https://emberjs.com/deprecations/v3.x#toc_jquery-apis",
+            for: "ember-source",
+          }
+        );
+
+        if (this.element) {
+          return sel ? jQuery(sel, this.element) : jQuery(this.element);
```

Comment:
Good call - I did some digging. It looks like the `.$()` function doesn't exist at all on the `Component` superclass in Ember 3.28. So this shim actually introduces the functionality, as well as deprecating it.

(This is lifted pretty much as-is from https://github.com/emberjs/ember-jquery/blob/master/vendor/jquery/component.dollar.js, with some slight tweaks to avoid triggering the 'ember global' deprecation)